src/node.js

```diff
@@ -489,6 +489,8 @@
       // on the way out, don't bother. it won't get fired anyway.
       if (process._exiting)
         return;
+      if (typeof callback !== 'function')
```

Comment:
I would move this `if` above the previous `if` for a more consistent throwing experience.